src/components/UserAutomations.tsx

```diff
@@ -1,50 +1,123 @@
 import { useSession } from "next-auth/react";
 import { type AutomationType } from "~/server/types/automation";
 import { api } from "~/utils/api";
+import { AiFillDelete, AiOutlineLoading } from "react-icons/ai";
+import { toast } from "react-hot-toast";
+import { useState } from "react";
 
 const UserAutomations = () => {
   const userAutomations = api.automation.getUserAutomations.useQuery();
   const deleteAutomation = api.automation.deleteAutomation.useMutation();
   const deleteWebHook = api.webhook.deleteWebhook.useMutation();
+  const [loading, setLoading] = useState(false);
 
   const { data: session } = useSession();
 
-  const handleDelete = async (automation: AutomationType) => {
-    const deleteWebHookAndReturnHookID = await deleteWebHook.mutateAsync({
-      hookID: Number(automation.webhookID),
-      accessToken: session?.user.accessToken,
-    });
-    if (deleteWebHookAndReturnHookID) {
-      deleteAutomation.mutate({ hookID: automation.webhookID });
+  const handleDelete = async (
+    automation: AutomationType,
+    e: React.FormEvent
+  ) => {
+    e.preventDefault();
+    setLoading(true);
+    if (loading) {
+      toast.dismiss();
+      toast.error("Stop spamming 😡");
     } else {
-      console.log("log out and log back in");
+      toast.dismiss();
+      toast.loading("Deleting Automation");
+      const deleteWebHookAndReturnHookID = await deleteWebHook.mutateAsync({
+        hookID: Number(automation.webhookID),
+        accessToken: session?.user.accessToken,
+      });
+
+      if (
+        deleteWebHookAndReturnHookID &&
+        deleteWebHookAndReturnHookID !== 404
```

Comment:
what does this variable `deleteWebHookAndReturnHookID` hold? nothing indicates that it is nullable or that it can hold an HTTP error code